File: backend/app/models/memory.py
from pydantic import BaseModel
from typing import List, Dict, Any, Optional
from datetime import datetime

class Memory(BaseModel):
    """Data model for AI agent memories"""
    id: Optional[str] = None
    content: str
    timestamp: datetime
    type: str
    tags: List[str]
    source: str
    metadata: Optional[Dict[str, Any]] = None

    class Config:
        # Allow arbitrary types
        arbitrary_types_allowed = True

File: frontend/src/types/memory.ts
export interface Memory {
  id: string;
  content: string;
  timestamp: Date;
  type: string;
  tags: string[];
  source: string;
  metadata: Record<string, any>;
}

export interface MemoryFormData {
  content: string;
  type: string;
  tags: string;
  source: string;
}